### DISCUSSION TITLE
Can't get UGS to work on Ubuntu

### DISCUSSION THREAD

**Discussion by mawoudenberg** [winder/Universal-G-Code-Sender#2158] [Help]
Hi,

I have my plasma cnc controlled by UGS on my windows computer, but I sometimes get freezes due to lots of EMI. I have an old single board computer lying around that I want to enclose in the Faraday box for all electronics to evoid the USB connection being lost by EMI.

I installed linux on it and downloaded the linux version of UGS and extracted. I installed java (default jre) so I have the latest version of OpenJDK RE running.
Now I go to the /bin folder where I find 2 .exe files, which both give "an error accurred while loading the archive", and another file that opens in a text editor. 

It's probably something simple, but as I'm very new to linux I don't know where to start looking. 

Any help would be greatly appreciated!

Thanks,
Maurits

**Comment by breiler** [comment]
You do not need to install Java to run UGS. Instructions for installing is available here:
https://github.com/winder/Universal-G-Code-Sender/wiki/Installing

You should be able to right click and choose "Run as program":
![Screenshot from 2023-02-17 10-44-35](https://user-images.githubusercontent.com/8962024/219610694-f649c7ba-61a0-4a5f-9cc2-e0d8da7092e8.png)
